packages/material-ui/src/CardHeader/CardHeader.test.js

```diff
@@ -31,14 +31,22 @@ describe('<CardHeader />', () => {
       expect(title).to.have.class(typographyClasses.h5);
     });
 
-    it('should render the subheader as body1 secondary text', () => {
+    it('should render the subheader as body1 secondary text', function test() {
+      if (/jsdom/.test(window.navigator.userAgent)) {
+        // Enable once https://github.com/jsdom/jsdom/issues/2898 is resolved
+        this.skip();
+      }
+
       const cardHeader = render(<CardHeader title="Title" subheader="Subheader" />).container
         .firstChild;
       const wrapper = cardHeader.firstChild;
       const subheader = wrapper.childNodes[1];
       expect(subheader).to.have.class(typographyClasses.root);
       expect(subheader).to.have.class(typographyClasses.body1);
-      expect(subheader).to.have.class(typographyClasses.colorTextSecondary);
+      // text secondary color
+      expect(subheader).toHaveComputedStyle({
+        color: 'rgba(0, 0, 0, 0.54)',
+      });
```

Comment:
I don't think we need to test the color here. A visual regression test would be more adequate here. Same applies to the other tests.

Reply:
I was thinking of the opposite, visual regressions sounds too slow and not accurate enough for this case. I'm also worried about the complexity it creates for changing the value of the theme later on. It's not very descriptive.

Maybe we should have a way to intercept the generated styles, before injected in the DOM to assert the value?

Reply:
> visual regressions sounds too slow and not accurate enough for this case

Can you confirm that?

>  I'm also worried about the complexity it creates for changing the value of the theme later on. It's not very descriptive.

What do you mean by "it"?

> Maybe we should have a way to intercept the generated styles, before injected in the DOM to assert the value?

Why?

Reply:
I think that we would be ideal is to be able to assert that the color === theme.palette.text.secondary.

Reply:
This was my first thought, but there is an issue as the colors in the `toHaveComputedStyles` always have rgb form. 

Apart from on visual tests, we could add custom theme here and check the value from that theme. That way future changes on the theme won't affect the tests.

Reply:
If it was working in JSDOM it would be even better. I was thinking of something similar to https://github.com/styled-components/jest-styled-components.

Reply:
So should we go with custom theme, or depend on visual tests here? I will check whether we already have a screen for testing all of these scenarios